extension/src/testStatusBar.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { window, StatusBarAlignment, StatusBarItem } from "vscode";
+
+import * as Commands from "./commands";
+import { TestLevel, TestStatus, TestSuite } from "./protocols";
+
+export class TestStatusBar {
+    private statusBar: StatusBarItem = window.createStatusBarItem(StatusBarAlignment.Left, Number.MIN_VALUE);
+
+    public update(tests: TestSuite[]): void {
+        let failedCount: number = 0;
+        let passedCount: number = 0;
+        const testMethods: TestSuite[] = tests.map((t) => t.level === TestLevel.Method ? [t] : t.children)
+                                 .reduce((a, b) => a.concat(b));
+        for (const t of testMethods) {

```

Comment:
failedCount = testMethods.reduce(m => t.result.status === TestStatus.Fail ? 1 : 0);

Reply:
`map` and `reduce` makes code neat, but for the scenario, we have to loop twice, however using for-loop we only need one loop.